src/bee.ts

```diff
@@ -139,12 +156,19 @@ export class Bee {
     options?: FileUploadOptions,
   ): Promise<Reference> {
     assertBatchId(postageBatchId)
+    assertFileData(data)
+
+    if (options) assertFileUploadOptions(options)
+
+    if (name && typeof name !== 'string') {
+      throw new TypeError('name has to be string or undefined!')
+    }
 
     if (isFile(data)) {
       const fileData = await fileArrayBuffer(data)
-      const fileName = name || data.name
+      const fileName = name ?? data.name
       const contentType = data.type
-      const fileOptions = options !== undefined ? { contentType, ...options } : { contentType }
+      const fileOptions = options ? { contentType, ...options } : { contentType }
```

Comment:
This would also work:
```suggestion
      const fileOptions = { contentType, ...options }
```

Reply:
Heh, true!